addon/components/cf-field/input/powerselect.js

```diff
@@ -0,0 +1,89 @@
+import Component from "@ember/component";
+import { computed } from "@ember/object";
+import { inject as service } from "@ember/service";
+import layout from "../../../templates/components/cf-field/input/powerselect";
+
+/**
+ * Dropdown component for the single and multiple choice question type
+ *
+ * @class CfFieldInputPowerSelectComponent
+ * @argument {Field} field The field for this input type
+ */
+export default Component.extend({
+  layout,
+  tagName: "",
+  intl: service(),
+
+  multiple: computed("field.question.__typename", function() {
+    return this.get("field.question.__typename").startsWith("Multiple");
+  }),
+
+  choices: computed(
+    "multiple",
+    "field.question.{choiceOptions,multipleChoiceOptions}.edges",
+    function() {
+      let options = this.get("multiple")
+        ? this.get("field.question.multipleChoiceOptions")
+        : this.get("field.question.choiceOptions");
+
+      return options.edges.map(edge => edge.node);
+    }
+  ),
+
+  selected: computed(
+    "field.answer.{_valueKey,listValue,stringValue}",
+    function() {
+      let key = this.get("field.answer._valueKey");
+      let path = `field.answer.${key}`;
+      let answers = this.get(path);
+      let selection = null;
+
+      if (answers !== null) {
+        selection = this.get("choices").filter(choice => {
+          return answers.includes(choice.slug);
+        });
+
+        if (key === "stringValue") {
+          selection = selection[0];
+        }
+      }
+
+      return selection;
+    }
+  ),
+
+  componentName: computed("multiple", function() {
+    let name = "power-select";
+
+    if (this.get("multiple")) {
+      name += "-multiple";
+    }
+
+    return name;
```

Comment:
```js
return this.get("multiple") ? "power-select-multiple" : "power-select"
```
looks more readable and concise to me, even though the strings have a large part in common.

Reply:
Makes sense.